mozilla/fx-private-relay
# Improvements for phone masking

This PR fixes MPP-2280  

# New feature description

MPP-2279 actually deals with the drop-down for city/name when choosing a Relay Number - that is not handled here. I mistakenly worked on 2280 + a number of items caught during that work - listed below:

- Real phone number input takes only numbers now 
- Real phone number input can take Canadian numbers and US numbers (limited to these countries)
- Formatting for error message logic added into `formatPhone` (validates length and deals with those numbers)
- Simple test added for `formatPhone`
- New library added in older PR to format phones used to deal with phone formatting. 
- We limit the max length of this input
- Relay number suggestions step was showing blank when running out of numbers. Now it pulls a new list once we run out. 

# Screenshot (if applicable)

Canadian Number:
<img width="817" alt="image" src="https://user-images.githubusercontent.com/3924990/186199789-df0c3b34-6d11-48c9-89ac-6e536bd05cfd.png">

US Number: 
<img width="662" alt="image" src="https://user-images.githubusercontent.com/3924990/186199887-13f650de-c4d7-46ca-8852-9e1085e513de.png">

# How to test
Go to the /phone page. 
Try entering a CA number +1250xxxxxxx
Try entering a US number +1909xxxxxxx
Try entering letters or non-digit characters
Confirm a real number and get to the relay number step - spam click the `get more options` button - once it runs out, it should load more suggestions. 


# Checklist
